File: src/com/sterndu/data/transfer/DatatransferServerSocket.java
package com.sterndu.data.transfer;

import java.io.IOException;
import java.net.*;

public abstract class DatatransferServerSocket extends ServerSocket {
	public DatatransferServerSocket() throws IOException {}

	public DatatransferServerSocket(int port) throws IOException {
		super(port);
	}

	public DatatransferServerSocket(int port, int backlog) throws IOException {
		super(port, backlog);
	}

	public DatatransferServerSocket(int port, int backlog, InetAddress bindAddr) throws IOException {
		super(port, backlog, bindAddr);
	}

	public DatatransferServerSocket(SocketImpl impl) {
		super(impl);
	}

	@Override
	public abstract Socket accept();
}

File: src/com/sterndu/data/transfer/DatatransferServerSocket.kt
@file:JvmName("DatatransferServerSocket")
package com.sterndu.data.transfer

import java.io.IOException
import java.net.InetAddress
import java.net.ServerSocket
import java.net.Socket

abstract class DatatransferServerSocket : ServerSocket {

	@Throws(IOException::class)
	constructor()
	@Throws(IOException::class)
	constructor(port: Int) : super(port)
	@Throws(IOException::class)
	constructor(port: Int, backlog: Int) : super(port, backlog)
	@Throws(IOException::class)
	constructor(port: Int, backlog: Int, bindAddr: InetAddress) : super(port, backlog, bindAddr)

	@Throws(IOException::class)
	abstract override fun accept(): Socket
}
